src/module.ts

```diff
@@ -73,14 +71,37 @@ const imageModule: Module<ModuleOptions> = async function imageModule (moduleOpt
     }
   })
 
-  addServerMiddleware({
-    path: '/_ipx',
-    handle: createIPXMiddleware({
-      dir: options.dir,
+  // Only add IPX server middleware if the static/ipx provider is used
+  if (
+    (options.provider === 'static' && nuxt.options.dev) ||
+    options.provider === 'ipx' ||
+    Object.keys(options.providers).includes('ipx')
+  ) {
+    const rootDir = nuxt.options.rootDir
+    const ipxOptions = {
+      dir: relative(rootDir, options.dir),
       domains: options.domains,
       sharp: options.sharp
-    })
-  })
+    }
+
+    // In development, add IPX middleware directly
+
+    const hasUserProvidedMiddleware = !!nuxt.options.serverMiddleware.find((mw: { path: string }) => mw.path && mw.path.startsWith('/_ipx'))
+
+    if (!hasUserProvidedMiddleware) {
+      const { createIPX, createIPXMiddleware } = await import('ipx')
+
+      const ipx = createIPX(ipxOptions)
+      addServerMiddleware({
+        path: '/_ipx',
+        handle: createIPXMiddleware(ipx)
```

Comment:
Can we instead generate `.nuxt/image/ipx.js` and register it by path? This makes it working for nitro prod build and making manual instructions easier before 2.16. (for development we can keep doing this until nitro has a better way for shared variable)